src/core/sock/sockinfo_tcp.cpp

```diff
@@ -1393,7 +1393,7 @@ err_t sockinfo_tcp::ip_output(struct pbuf *p, struct tcp_seg *seg, void *v_p_con
     void *cur_end;
 
     if (unlikely(flags & XLIO_TX_PACKET_REXMIT)) {
-        if (unlikely(inspect_socket_error_state(reinterpret_cast<const mem_buf_desc_t *>(p),
+        if (unlikely(inspect_socket_error_state(reinterpret_cast<const mem_buf_desc_t *>(seg->p),
```

Comment:
I think we should not ignore the investigation of why this bug caused the shrink w/o FIN bug. 
it's not obvious to me - and I think the effort will either uncover more bugs, or lead to gaining a very important insight.